### ISSUE TITLE
creating bundles

### ISSUE THREAD

**Issue by mikaelkaron** [requirejs/requirejs#292]
I'm writing a framework for web development. Individual parts of the framework are split into submodules like so:

```
.
└── lib
    ├── ...
    ├── troopjs-core
    ├── troopjs-jquery
    └── troopjs-requirejs
```

For convenience I'd like to bundle all of the parts together, so I created `troopjs-bundle` that when run through r.js creates a file with all of the modules defined.

In a project that uses troopjs-bundle I define my config like so:

``` javascript
require({
    "baseUrl" : "js",
    "paths" : {
        "jquery" : "lib/jquery/dist/jquery",
        "troopjs-bundle" : "lib/troopjs-bundle/dist/troopjs-bundle"
    },
    "deps" : [ "troopjs-bundle" ]
}, [ "jquery" ], function App(jQuery) {
});
```

this works ok (for now) with one exception. I can depend on modules defined in the bundle (like `troopjs-core/component/widget`) in other modules (let's say I create `widget/test.js` that dependes on `troopjs-core/component/widget`) - but not from `app.js` itself.

Basically if I replace

``` javascript
}, [ "jquery" ], function App(jQuery) {
```

with

``` javascript
}, [ "jquery", "troopjs-core/component/widget" ], function App(jQuery, Widget) {
```

it fails, but if I have `widget/test` that requires `troopjs-core/component/widget` and I do

``` javascript
}, [ "jquery", "widget/test" ], function App(jQuery, Test) {
```

it works fine.
